### ISSUE TITLE
can't use Grails mocking support in setupSpec

### ISSUE THREAD

**Issue by robfletcher** [spockframework/spock#236]
Originally reported on Google Code with ID 114

```
The internal fields in UnitSpec are not shared, which prevents certain functions working
from setupSpec.
```

Reported by `ldaley` on 2010-07-28 07:25:14

**Comment by frozenspider** [comment]
Not sure whether it's the same bug or not, but Grails domain mocking can't be used in `setupSpec` with Grails 2.1.0 + Spock 0.7
Base class is as follows:
```
@TestMixin(DomainClassUnitTestMixin)
abstract class DomainMockingSpecBase extends Specification {
  ...
  def setupSpec() {
    ...
  }
```
* If the domains are mocked via `@Mock`, calling `user.save(failOnError: true)` reports
```
|  groovy.lang.MissingMethodException: No signature of method: User.save() is applicable for argument types: () values: []
Possible solutions: save(), save(boolean), save(java.util.Map), wait(), any(), wait(long)
```
(i.e. same as without `@Mock` at all)
* If the domains are mocked via `mockDomain(User)`, it acts in the weird way:
```
        user1 = createUser("user1", "fn1", "ln1", "e@ma1.il")
        user2 = createUser("user2", "fn2", "ln2", "e@ma2.il")
        assert user1.id == 1
        assert user2.id == 2
```
Second assertion fails because `user2.id == 1` for some reason.

Both issues are only relevant to `setupSpec`, not `setup`, but we want the shared domains to be accessible in the `where` block so we need some way to create them before `where` gets evaluated.

**Comment by philipbed** [comment]
@frozenspider did you ever find a fix? I am encounering the same problem

**Comment by frozenspider** [comment]
No, but we came up with a workaround - create entities in `setup` and only use strings/IDs in `where`, fetching entities in test case code itself.

**Comment by philipbed** [comment]
Okay, thank you. I will try that.
